CMakeLists.txt

```diff
@@ -110,26 +104,44 @@ if (WIN32)
         string(REPLACE "/MDd" "/MTd" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
     endif()
 else()
-    if (APPLE)
-        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
-    else ()
+    target_link_libraries (OpenCL PRIVATE ${CMAKE_THREAD_LIBS_INIT})
+    if (NOT APPLE)
         set_target_properties (OpenCL PROPERTIES LINK_FLAGS "-Wl,--version-script -Wl,${CMAKE_CURRENT_SOURCE_DIR}/loader/linux/icd_exports.map")
         if (OPENCL_ICD_LOADER_PIC)
             set_target_properties(OpenCL PROPERTIES POSITION_INDEPENDENT_CODE ON)
         endif ()
-        target_link_libraries (OpenCL ${CMAKE_THREAD_LIBS_INIT})
     endif ()
 endif ()
 
-include_directories (${OPENCL_ICD_LOADER_HEADERS_DIR})
-add_definitions (-DCL_TARGET_OPENCL_VERSION=300)
+find_package(OpenCLHeaders CONFIG NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH)
```

Comment:
why this? this break the search when the headers(and the cmake files) is installed in the system

~~~
-- Could NOT find OpenCLHeaders (missing: OpenCLHeaders_DIR)
~~~

change to `REQUIRED` works again

Reply:
seems set `-DCMAKE_FIND_USE_CMAKE_PATH=TRUE` do the trick instead of edit the cmakelist.txt

greetings and sorry for the noise